### PR TITLE
remove-api-key updated

### PR DESCRIPTION
**Pull request by jesusgonzalez12**
This version introduces improvements on how a user can remove an API key using the pyc8 SDK.
Only changes to the pyc8 code have been introduced.
For more details, you can check the Kanban board.

### CODE REVIEW

File: docs/sdks/remove-api-key.md [hunk 1]
```diff
@@ -32,10 +32,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
...
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check wether the chosen_api_key exists or not, comparing it to the list elements
+for apiElement in apiKeyIds:
+    if apiElement == CHOSEN_API_KEY_ID:
+        API_ACTIVE = True
+        break
 
-print(remove)
+# Checks if the API already exists and acts accordingly
```
Comment (jamesm-macrometa) [inline docs/sdks/remove-api-key.md]:
```suggestion
# Checks the specified API key
```

If I understand correctly, this step merely checks the API key, and the next one removes it. Correct?

File: docs/sdks/remove-api-key.md [hunk 2]
```diff
@@ -32,10 +32,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
...
+if API_ACTIVE:
+    # Remove the chosen API key
+    try:
+        print("Remove API Key: ", client.remove_api_key(CHOSEN_API_KEY_ID))
+    except Exception as err:
+        print("Error removing the API key")
+        print(err)
+else:
+    # Display message that the eky does not exists
```
Comment (jamesm-macrometa) [inline docs/sdks/remove-api-key.md]:
```suggestion
    # Displays a message that the key does not exist
```

File: docs/sdks/remove-api-key.md [hunk 3]
```diff
@@ -32,10 +32,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
...
+# Check wether the chosen_api_key exists or not, comparing it to the list elements
+for apiElement in apiKeyIds:
+    if apiElement == CHOSEN_API_KEY_ID:
+        API_ACTIVE = True
+        break
 
-print(remove)
+# Checks if the API already exists and acts accordingly
+if API_ACTIVE:
+    # Remove the chosen API key
```
Comment (jamesm-macrometa) [inline docs/sdks/remove-api-key.md]:
```suggestion
    # Removes the chosen API key
```

File: docs/sdks/remove-api-key.md [hunk 4]
```diff
@@ -32,10 +32,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
                         email='nemo@nautilus.com', password='xxxxx',
                         geofabric='_system')
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check wether the chosen_api_key exists or not, comparing it to the list elements
```
Comment (jamesm-macrometa) [inline docs/sdks/remove-api-key.md]:
```suggestion
# Checks if `chosen_api_key` exists
```

What do we mean by "comparing it to the list elements"?

**Review by jamesm-macrometa — CHANGES_REQUESTED** [review]
A few minor changes, but looks good!

File: docs/sdks/remove-api-key.md (hunk 4 above)
Comment (jesusgonzalez12) [reply docs/sdks/remove-api-key.md]: Sorry, I missed adding a few lines of code. Fixed.

A list of all API key IDs a user has is saved in a variable which is used to make a comparison check later.

**Comment by jesusgonzalez12** [comment]
Thanks for the review.
Please, check again the element list topic.

File: docs/sdks/remove-api-key.md (hunk 4 above)
Comment (jesusgonzalez12) [reply docs/sdks/remove-api-key.md]: I have updated the comment to improve semantics.

File: docs/sdks/remove-api-key.md [hunk 5]
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
...
                         geofabric='_system')
+                        
+# Get a list of all the API key ID's names
+apiKeyIds = []
+for api in client.list_all_api_keys():
+    apiKeyIds.append(api["keyid"])
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check wether the chosen_api_key exists or not, comparing it to the API key list
```
Comment (diana-macrometa) [inline docs/sdks/remove-api-key.md]:
```suggestion
# Check whether the chosen_api_key exists or not, comparing it to the API key list
```

File: docs/sdks/remove-api-key.md [hunk 6]
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
...
+    if apiElement == CHOSEN_API_KEY_ID:
+        API_ACTIVE = True
+        break
 
-print(remove)
+# Checks the specified API key
+if API_ACTIVE:
+    # Removes the chosen API key
+    try:
+        print("Remove API Key: ", client.remove_api_key(CHOSEN_API_KEY_ID))
```
Comment (diana-macrometa) [inline docs/sdks/remove-api-key.md]:
```suggestion
        print("Remove API key: ", client.remove_api_key(CHOSEN_API_KEY_ID))
```

Comment (jesusgonzalez12) [reply docs/sdks/remove-api-key.md]: Thanks for the suggestions @diana-macrometa. Updated with suggestions.

File: docs/sdks/remove-api-key.md [hunk 7]
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
...
                         geofabric='_system')
+                        
+# Get a list of all the API key ID's names
+apiKeyIds = []
+for api in client.list_all_api_keys():
+    apiKeyIds.append(api["keyid"])
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check whether the chosen_api_key exists or not, comparing it to the API key list
```
Comment (diana-macrometa) [inline docs/sdks/remove-api-key.md]: Why is chosen_api_key lowercase here and uppercase in line 42?

File: docs/sdks/remove-api-key.md [hunk 8]
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
...
+if API_ACTIVE:
+    # Removes the chosen API key
+    try:
+        print("Remove API key: ", client.remove_api_key(CHOSEN_API_KEY_ID))
+    except Exception as err:
+        print("Error removing the API key")
+        print(err)
+else:
+    # Displays a message that the key does not exist
+    print("Can't remove a non-existent key")
```
Comment (diana-macrometa) [inline docs/sdks/remove-api-key.md]:
```suggestion
    print("Can't remove a non-existent key.")
```

Comment (diana-macrometa) [reply docs/sdks/remove-api-key.md]: Be sure to finish sentences with periods. Apply throughout.

**Comment by diana-macrometa** [comment]
@jesusgonzalez12, click the circling arrows next to a reviewer to request another review.

File: docs/sdks/remove-api-key.md (hunk 7 above)
Comment (jesusgonzalez12) [reply docs/sdks/remove-api-key.md]: Hi Diana. PEP8 standard (and JS standard is similar) states that constants should be in upper-case.

We recently changed our code to be as PEP8 compliant as possible.
And certainly, that comment line was bypassed while making the conversion.
Will fix it shortly.

File: docs/sdks/remove-api-key.md (hunk 8 above)
Comment (jesusgonzalez12) [reply docs/sdks/remove-api-key.md]: Thanks Diana.

This writing style is not in the original files.
Good to have it as a way to increase consistency on the docs.
Fixed.

File: docs/sdks/remove-api-key.md [hunk 9]
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
...
                         geofabric='_system')
+                        
+# Get a list of all the API key ID's names
+apiKeyIds = []
+for api in client.list_all_api_keys():
+    apiKeyIds.append(api["keyid"])
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Check whether the CHOSEN_API_KEY_ID exists or not, comparing it to the API key list
```
Comment (jamesm-macrometa) [inline docs/sdks/remove-api-key.md]:
```suggestion
# Checks if the API key exists
```

**Review by jamesm-macrometa — APPROVED** [review]

File: docs/sdks/remove-api-key.md [hunk 10]
```diff
@@ -31,11 +31,29 @@ from c8 import C8Client
...
+apiKeyIds = []
+for api in client.list_all_api_keys():
+    apiKeyIds.append(api["keyid"])
 
-# Remove api key
-remove = client.remove_api_key('id1')
+# Checks if the API key exists
+for apiElement in apiKeyIds:
+    if apiElement == CHOSEN_API_KEY_ID:
+        API_ACTIVE = True
```
Comment (koshy-thomas) [inline docs/sdks/remove-api-key.md:47]: @jesusgonzalez12 I guess you forgot to declare CHOSEN_API_KEY_ID and API_ACTIVE on top after import C8Client.
Example:
API_ACTIVE = False
CHOSEN_API_KEY_ID = "YourApiKey"

Comment (koshy-thomas) [reply docs/sdks/remove-api-key.md:47]: @jesusgonzalez12 The same is missing I guess in "docs/sdks/create-api-key.md" as well you can add the declaration of these two variables there too.

Comment (jesusgonzalez12) [reply docs/sdks/remove-api-key.md:47]: Thanks @koshy-thomas. You're right, thanks for checking. Making changes.

PS: for this I have a single file that adds the key if the ID does not exists while removing it If the key already exists. So translating that to two separate files I had to change a few prints and picky-choose what I needed for each example. As I can see, I have room for improvement on that department.

Changes here are complete. Will now check PR 205.

**Pull request merged**